### PR TITLE
Fixed index in Stack peek

### PR DESCRIPTION
**Pull request by sebastianbaginski**
Added `top` function

### CODE REVIEW

**Comment by davidhesselbom** [comment]
Thanks. Feel free to add a test for this at some point...

**Comment by sebastianbaginski** [comment]
Sure.

File: source/sdk/structs/Stack.ooc [hunk 1]
```diff
@@ -23,13 +23,16 @@ Stack: class <T> {
...
 		version(safe)
 			if (index < 0)
 				Exception new(This, "Trying to peek(%d)! index must be >= 1 <= size" format(index)) throw()
 			else if (index >= this size)
 				Exception new(This, "Trying to peek(%d) a stack of size %d" format(index, this size)) throw()
-		this _data[index]
+		this _data[this size - index - 1]
 	}
 	//TODO: remove this function, exists only for compatibility with current codebase
 	empty?: func -> Bool {
```
Comment (marcusnaslund) [inline source/sdk/structs/Stack.ooc]: You only need to change one file (`Exception.ooc`) to get rid of this, please do.

Comment (marcusnaslund) [inline source/sdk/structs/Stack.ooc]: This is only used in `Exception.ooc`. I'll fix that when I add a test.

**Comment by marcusnaslund** [comment]
Looks good @davidhesselbom

File: source/sdk/structs/Stack.ooc (hunk 1 above)
Comment (sebastianbaginski) [inline source/sdk/structs/Stack.ooc]: OK, I just cleaned it up

Comment (marcusnaslund) [inline source/sdk/structs/Stack.ooc]: Fair enough, just didn't want to cause conflicts with your Exception PR.

File: source/sdk/structs/Stack.ooc [hunk 2]
```diff
@@ -23,17 +23,16 @@ Stack: class <T> {
 				Exception new(This, "Trying to pop an empty stack.") throw()
 		this _data[--this _size]
 	}
+	top: func -> T {
```
Comment (davidhesselbom) [inline source/sdk/structs/Stack.ooc]: Should this be a function, or a property?

Comment (sebastianbaginski) [inline source/sdk/structs/Stack.ooc]: I don't know, you can decide.
It *will* be a function under the hood, anyway :)

File: source/sdk/structs/Stack.ooc (hunk 1 above)
Comment (sebastianbaginski) [inline source/sdk/structs/Stack.ooc]: Thanks. I rebase all my PRs after any change in kean anyway. I think I will create a script for that...

File: source/sdk/structs/Stack.ooc (hunk 2 above)
Comment (davidhesselbom) [inline source/sdk/structs/Stack.ooc]: That's true. I suggest `top ::= this peek()`.

Comment (sebastianbaginski) [inline source/sdk/structs/Stack.ooc]: OK, updated.

File: source/sdk/lang/Exception.ooc [hunk 3]
```diff
@@ -56,7 +56,7 @@ _markStackFrame: inline func (frame: StackFrame) {
 }
 
 _cleanupStackFrames: inline func {
-  while (stackFramesToDelete empty?() == false) {
+  while (stackFramesToDelete isEmpty == false) {
```
Comment (davidhesselbom) [inline source/sdk/lang/Exception.ooc]: Minor gripe: `while (!stackFramesToDelete isEmpty) {` (and below)

Comment (davidhesselbom) [inline source/sdk/lang/Exception.ooc]: I could also imagine a `notEmpty` property.

Comment (sebastianbaginski) [inline source/sdk/lang/Exception.ooc]: Hmm so should it be `isEmpty && notEmpty`, or `isEmpty && isNotEmpty` ? :)

Comment (davidhesselbom) [inline source/sdk/lang/Exception.ooc]: Buuuut apparently Simon doesn't, so don't add one. :)

Comment (davidhesselbom) [inline source/sdk/lang/Exception.ooc]: Neither, just go with a `!`. Sorry I brought it up

Comment (sebastianbaginski) [inline source/sdk/lang/Exception.ooc]: Hehe ok, fixed.

**Pull request merged**